src/views/AccessControl.tsx

```diff
@@ -127,26 +127,30 @@ export const AccessControl = () => {
     const saveKey = 'saving';
     useEffect(() => {
         if (savedRule.loading) {
-            message.loading({ content: 'Saving...', key: saveKey, duration: 0, style: styleNotification });
+            message.loading({ content: 'Saving...', key: saveKey, duration: 0, style: styleNotification })
         } else if (savedRule.success) {
-            message.success({ content: 'Rule saved with success!', key: saveKey, duration: 2, style: styleNotification });
-            dispatch(ruleActions.setSetupNewRuleVisible(false));
-            dispatch(ruleActions.setSavedRule({ ...savedRule, success: false }));
+            message.success({ content: 'Rule has been successfully updated.', key: saveKey, duration: 2, style: styleNotification });
+            dispatch(ruleActions.setSetupNewRuleVisible(false))
+            dispatch(ruleActions.setSavedRule({ ...savedRule, success: false }))
+            dispatch(ruleActions.resetSavedRule(null))
         } else if (savedRule.error) {
-            message.error({ content: 'Error! Something wrong to create key.', key: saveKey, duration: 2, style: styleNotification  });
-            dispatch(ruleActions.setSavedRule({ ...savedRule, error: null }));
+            message.error({ content: 'Failed to update rule. You might not have enough permissions.', key: saveKey, duration: 2, style: styleNotification  });
+            dispatch(ruleActions.setSavedRule({ ...savedRule, error: null }))
```

Comment:
Codacy has a fix for the issue: [Replace `······dispatch(ruleActions.setSavedRule({·...savedRule,·error:·null·}))` with `});⏎······dispatch(ruleActions.setSavedRule({·...savedRule,·error:·null·}));`](https://app.codacy.com/gh/netbirdio/dashboard/pullRequest?prid=9560373)
```suggestion
      });
```